core/sdk-core/src/main/java/software/amazon/awssdk/core/progress/snapshot/ProgressSnapshot.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.progress.snapshot;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalLong;
+import java.util.concurrent.TimeUnit;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.annotations.ThreadSafe;
+import software.amazon.awssdk.core.SdkResponse;
+
+@Immutable
+@ThreadSafe
+@SdkPublicApi
+public interface ProgressSnapshot {
+    /**
+     * The total number of bytes that have been sent or received so far.
+     */
+    long transferredBytes();
+
+    /**
+     * Time at which transaction started
+     */
+    Instant startTime();
+
+    /**
+     * Elapsed time since the start of the transaction
+     */
+    Duration elapsedTime();
+
+    /**
+     * If transaction size is known, estimate time remaining for transaction completion
+     */
+    Optional<Duration> estimatedTimeRemaining();
+
+    /**
+     * The SDK response, or {@link Optional#empty()} if unknown.
+     */
+    Optional<SdkResponse> sdkResponse();
```

Comment:
This was on the context objects. Do we need it here as well?

Reply:
I thought it makes sense to read it from the snapshot in case we are downloading a larger payload that might perform a download in multiple stages. Felt like it should be associated with a snapshot and used like here, as TM uses it
https://github.com/aws/aws-sdk-java-v2/blob/0b9b0735c447af78316e5d7ef960e2b237f2f073/services-custom/s3-transfer-manager/src/main/java/software/amazon/awssdk/transfer/s3/internal/model/DefaultFileDownload.java#L70